foreshadow/utils/default_estimator_factory.py

```diff
@@ -0,0 +1,121 @@
+"""Estimator factory and supported estimators."""
+
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.linear_model import LinearRegression, LogisticRegression
+from sklearn.neural_network import MLPClassifier, MLPRegressor
+from sklearn.svm import LinearSVC, LinearSVR
+
+
+class EstimatorFamilyMixin:
+    """Mixin class that allows an instance to retrieve an estimator.
+
+    Retrieve is based on the type of the problem (classification or
+    regression).
+    """
+
+    def get_estimator_by_type(self, problem_type):
+        """Retrieve an estimator based on the problem type.
+
+        Args:
+            problem_type: either classification or regression
+
+        Returns:
+            an estimator of a particular model family
+
+        """
+        return self.estimators[problem_type]
+
+
+class LinearEstimatorFamily(EstimatorFamilyMixin):
+    """Linear model estimator family.
+
+    Estimators include LinearRegression and LogisticRegression.
+    """
+
+    def __init__(self):
+        self.estimators = {
+            "classification": LogisticRegression(),
+            "regression": LinearRegression(),
```

Comment:
We use 'classification' and 'regression' multiple times, which make it worthwhile to create two module-level constants. Then these constants can be used to provide choices for command-line console.